core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -519,41 +483,11 @@ object DumpLogSegments {
       json
     }
 
-    private def prepareValue(message: Message, version: Short): String = {
-      val messageAsJson = message match {
-        case m: OffsetCommitValue =>
-          OffsetCommitValueJsonConverter.write(m, version)
-        case m: GroupMetadataValue =>
-          prepareGroupMetadataValue(m, version)
-        case m: ConsumerGroupMetadataValue =>
-          ConsumerGroupMetadataValueJsonConverter.write(m, version)
-        case m: ConsumerGroupPartitionMetadataValue =>
-          ConsumerGroupPartitionMetadataValueJsonConverter.write(m, version)
-        case m: ConsumerGroupMemberMetadataValue =>
-          ConsumerGroupMemberMetadataValueJsonConverter.write(m, version)
-        case m: ConsumerGroupTargetAssignmentMetadataValue =>
-          ConsumerGroupTargetAssignmentMetadataValueJsonConverter.write(m, version)
-        case m: ConsumerGroupTargetAssignmentMemberValue =>
-          ConsumerGroupTargetAssignmentMemberValueJsonConverter.write(m, version)
-        case m: ConsumerGroupCurrentMemberAssignmentValue =>
-          ConsumerGroupCurrentMemberAssignmentValueJsonConverter.write(m, version)
-        case m: ConsumerGroupRegularExpressionValue =>
-          ConsumerGroupRegularExpressionValueJsonConverter.write(m, version)
-        case m: ShareGroupMetadataValue =>
-          ShareGroupMetadataValueJsonConverter.write(m, version)
-        case m: ShareGroupPartitionMetadataValue =>
-          ShareGroupPartitionMetadataValueJsonConverter.write(m, version)
-        case m: ShareGroupMemberMetadataValue =>
-          ShareGroupMemberMetadataValueJsonConverter.write(m, version)
-        case m: ShareGroupTargetAssignmentMetadataValue =>
-          ShareGroupTargetAssignmentMetadataValueJsonConverter.write(m, version)
-        case m: ShareGroupTargetAssignmentMemberValue =>
-          ShareGroupTargetAssignmentMemberValueJsonConverter.write(m, version)
-        case m: ShareGroupCurrentMemberAssignmentValue =>
-          ShareGroupCurrentMemberAssignmentValueJsonConverter.write(m, version)
-        case m: ShareGroupStatePartitionMetadataValue =>
-          ShareGroupStatePartitionMetadataValueJsonConverter.write(m, version)
-        case _ => throw new IllegalStateException(s"Message value ${message.getClass.getSimpleName} is not supported.")
+    private def prepareValue(message: ApiMessage, version: Short): String = {
+      val messageAsJson = if (message.apiKey == GroupCoordinatorRecordType.GROUP_METADATA.id) {
+        prepareGroupMetadataValue(message.asInstanceOf[GroupMetadataValue], version)
```

Comment:
The generated code is unable to parse the subscription. Therefore, we need to implement specific handling for `GroupMetadataValue`. If `GroupCoordinatorRecordJsonConverters.writeRecordValueAsJson` is exclusively used by `DumpLogSegments`, this approach should be acceptable. However, if this function is utilized in other contexts, I am concerned that it might generate JSON output that differs from other tools or logs.

Reply:
Yeah, but this is true with and without this change. `GroupMetadataValue` is annoying because it requires a special handling.